src/lib/annotations/__tests__/AnnotationThread-test.js

```diff
@@ -174,6 +174,54 @@ describe('lib/annotations/AnnotationThread', () => {
         });
     });
 
+    describe('updateLocalAnnotationFromServerResponse()', () => {
+        let annotationService;
+
+        beforeEach(() => {
+            annotationService = {
+                create: () => {}
+            };
+
+            thread = new AnnotationThread({
+                annotatedElement: document.querySelector('.annotated-element'),
+                annotations: [],
+                annotationService,
+                fileVersionId: '1',
+                location: {},
+                threadID: '2',
+                threadNumber: '1',
+                type: 'point'
+            });
+
+            stubs.create = sandbox.stub(annotationService, 'create');
+        });
+
+        it('should save annotation to thread if it does not exist in annotations array', () => {
+            const serverAnnotation = 'im a real annotation';
+            stubs.saveAnnotationToThread = sandbox.stub(thread, 'saveAnnotationToThread');
+            thread.updateLocalAnnotationFromServer(serverAnnotation, serverAnnotation);
+            thread.on('annotationcreateerror', () => {
+                Assert.fail('Annotation creation should not have failed');
+            });
+            expect(stubs.saveAnnotationToThread).to.be.called;
+        });
+
+        it('should overwrite a local annotation to the thread if it does exist as an associated annotation', () => {
+            const serverAnnotation = 'im a real annotation';
+            const replacementAnnotation = 'im a replacement annotation';
+            const isReplacementAnnotation = (annotation) => annotation === replacementAnnotation;
+            stubs.saveAnnotationToThread = sandbox.stub(thread, 'saveAnnotationToThread');
+            thread.annotations.push(serverAnnotation)
+            thread.updateLocalAnnotationFromServer(replacementAnnotation, serverAnnotation);
+            thread.on('annotationcreateerror', () => {
```

Comment:
Not sure the check on `'annotationcreateerror'` needs to be in the unit tests for `updateLocalAnnotationFromServer()`. Might make sense to move the portion of code called in the catch block into it's own method and move this check there.